artificer/src/test/java/net/technolords/tools/artificer/input/ArtifactResourceVisitorTest.java

```diff
@@ -14,16 +23,64 @@
     /**
      * Test case #1: Test classification of a resource.
      */
-    @Test
-    public void testClassificationOfResource() {
-        LOGGER.info("Todo: implementation of test case");
+    @DataProvider(name = "dataSetClassifyResource")
+    public Object[][] dataSetClassifyResource() {
+        return new Object[][] {
+                {"ArtificerImplTest.class","D:/commit2Learn/techno-tools/artificer/target/test-classes/net/technolords/tools/artificer/"},
+                {"abc","D:/commit2Learn/techno-tools/artificer/src/test/resources/data/"}
+        };
+    }
+
+    @Test(dataProvider = "dataSetClassifyResource")
+    public void testClassificationOfResource(String input, String location) {
+
+        Path pathToResourceLocation = FileSystems.getDefault().getPath(location);
+        LOGGER.debug("The path towards the input file '" + input + "' exists: " + Files.exists(pathToResourceLocation));
+
+        Analysis analysis = new Analysis();
+        analysis.setArtifactName(input);
+
+        // Create a resource reference linking to the file
+        Resource resource = new Resource();
+        resource.setPath(pathToResourceLocation);
+        resource.setName(input);
+
+        ArtifactResourceVisitor artifactResourceVisitor = new ArtifactResourceVisitor(analysis);
+        artifactResourceVisitor.classifyResource(resource);
+
+        Assert.assertTrue(true);
     }
 
     /**
      * Test case #2: Test adding a resource to a resource group.
      */
-    @Test
-    public void testAddResourceToClassificationGroup() {
-        LOGGER.info("Todo: implementation of test case");
+    @DataProvider(name = "dataSetResourceGroupClassification")
+    public Object[][] dataSetResourceGroupClassification() {
+        return new Object[][] {
+                {"ArtificerImplTest.class","D:/commit2Learn/techno-tools/artificer/target/test-classes/net/technolords/tools/artificer/",".class"},
+                {"abc","D:/commit2Learn/techno-tools/artificer/src/test/resources/data/","_classification_undefined_"},
+                {"artificer-1.0.0-SNAPSHOT.jar","D:/commit2Learn/techno-tools/artificer/src/test/resources/data/jars",".jar"}
+        };
     }
+
+    @Test(dataProvider = "dataSetResourceGroupClassification")
+    public void testAddResourceToClassificationGroup(String input, String location, String classification) {
```

Comment:
Consider to add a flag 'final' to the parameters and understand why this is a good practise. Hint, other test classes also use this. In addition, try to read about this on the internet as extension of your homework.